content/blog/2019/11/2019-11-21-welcome-to-the-matrix.adoc

```diff
@@ -0,0 +1,389 @@
+If we wanted to run only "linux" cells, we could use the following `exclude`: 
+[source, groovy]
+----
+exclude {
+    axis {
+        name 'PLATFORM'
+        notValues 'linux'
+    }
+}
+----
+====
+
+[source, groovy]
+----
+pipeline {
+    agent none
+    stages {
+        stage("BuildAndTest") {
+            matrix {
+                agent any
+                axes {
+                    axis {
+                        name 'PLATFORM'
+                        values "linux", "windows", "mac"
+                    }
+                    axis {
+                        name 'BROWSER'
+                        values "firefox", "chrome", "safari", "edge"
+                    }
+                }
+                excludes {
+                    exclude {
+                        axis {
+                            name 'PLATFORM'
+                            values 'linux'
+                        }
+                        axis {
+                            name 'BROWSER'
+                            values 'safari'
+                        }
+                    }
+                    exclude {
+                        axis {
+                            name 'PLATFORM'
+                            notValues 'windows'
+                        }
+                        axis {
+                            name 'BROWSER'
+                            values "edge"
+                        }
+                    }
+                }
+                stages {
+                    stage("Build") {
+                        steps {
+                            echo "Do Build for ${PLATFORM} - ${BROWSER}"
+                        }
+                    }
+                    stage("Test") {
+                        steps {
+                            echo "Do Test for ${PLATFORM} - ${BROWSER}"
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+----
+
+.Log output (trucated)
+[source]
+----
+...
+[Pipeline] stage
+[Pipeline] { (BuildAndTest)
+[Pipeline] parallel
+[Pipeline] { (Branch: Matrix - OS = 'linux', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'firefox')
+[Pipeline] { (Branch: Matrix - OS = 'linux', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'chrome')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'safari')
+[Pipeline] { (Branch: Matrix - OS = 'mac', BROWSER = 'safari')
+[Pipeline] { (Branch: Matrix - OS = 'windows', BROWSER = 'edge')
+...
+Do Build for linux - firefox
+...
+----
+
+== Controlling cell behavior at runtime
+
+Inside the `matrix` directive I can also add "per-cell" directives. 
+These are the same directives that I would add to a `stage` and they let me control the behavior of each cell of in the matrix.
```

Comment:
suggest changing to behavior of each cell in the matrix.

Reply:
```suggestion
These are the same directives that I would add to a `stage` and they let me control the behavior of each cell in the matrix.
```